asustor.c

```diff
@@ -357,27 +403,165 @@ static struct gpio_chip *find_chip_by_name(const char *name)
 }
 #endif
 
+/**
+ *	dmi_matches - check if dmi_system_id structure matches system DMI data
+ *	@dmi: pointer to the dmi_system_id structure to check
+ */
+// copied from dmi_matches() in Linux 6.11.2 drivers/firmware/dmi_scan.c where it's private (static)
+// with only one small change (dmi_val = dmi_get_system_info(s) instead of dmi_ident[s])
```

Comment:
hmm right, this might be a minor problem,
the license header of asustor.c is: `// SPDX-License-Identifier: GPL-2.0-or-later`
the license header of dmi_scan.c is `// SPDX-License-Identifier: GPL-2.0-only`

Reply:
In theory it's easy to just move that function into a `asustor_gpl2.c`.
In practice it would require me to understand that magic Makefile :-p
It currently seems to assume that each module has exactly one source file